src/Sylius/Bundle/ApiBundle/Serializer/ChannelPricingDenormalizer.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Sylius\Bundle\ApiBundle\Serializer;
+
+use Sylius\Bundle\ApiBundle\Helpers\CustomMoneyTransformerInitializer;
+use Sylius\Component\Core\Model\ChannelPricing;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+class ChannelPricingDenormalizer implements ContextAwareDenormalizerInterface
+{
+    /**
+     * @var DenormalizerInterface
+     */
+    private DenormalizerInterface $objectNormalizer;
+    private string $classType;
+
+    public function __construct(
+        DenormalizerInterface $objectNormalizer,
+        string $classType
+    )
+    {
+        $this->objectNormalizer = $objectNormalizer;
+        $this->classType = $classType;
+    }
+
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        $transformer = CustomMoneyTransformerInitializer::getPriceTransformer();
```

Comment:
We shouldn't create a new instance of `SyliusMoneyTransformer` via `CustomMoneyTransformerInitializer`. We must inject `SyliusMoneyTransformer` service and use it directly without a new layer